src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,48 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Encapsulate Deadline which is-a Task
+ */
+public class Deadline extends Task {
+
+    protected LocalDate by;
```

Comment:
```suggestion
    protected final LocalDate by;
```
Should you consider setting this attribute as final?

Reply:
I did not implement it as final as I may need to modify the date later on. For example, if a user wants to change the by date in deadline.